### ISSUE TITLE
Temporal / List not a member of type 'Amplify'

### ISSUE THREAD

**Issue by imikemiller** [aws-amplify/amplify-ios#581]
**Describe the bug**
Decoding JSON responses into Amplify models is failing with errors; `'Temporal' is not a member type of 'Amplify'` and `'List' is not a member type of 'Amplify'`. Model fields that are defined using Swift types decode successfully.

As the fields that fail to decode are optional the model ends up with nil values. Inspecting the response shows the errors as in the screen shot below.

**To Reproduce**

Following the documentation and using the codegen model: 

```
public struct Checkin: Model {
  public let id: String
  public var createdAt: Temporal.DateTime?
  public var updatedAt: Temporal.DateTime?
  public var dateOfBirth: Temporal.DateTime?
  public var user: User?
  public var tags: List<CheckinTag>?
  //redacted
}

_ = Amplify.API.query(request: .list(Checkin.self) { event in
            switch event {
            case .success(let result):
                switch result {
                case .success(let checkinList):
                    DDLogInfo("CheckinList: \(checkinList)")
                    self.dailySummaryPassthroughSubject.send(checkinList)
                case .failure(let error):
                    DDLogError("Failed result: \(error.errorDescription)")
                    self.dailySummaryPassthroughSubject.send(completion: .failure(error))
                }
            case .failure(let error):
                DDLogError("Failed event: \(error)")
                self.dailySummaryPassthroughSubject.send(completion: .failure(error))
            }
        }
```

**Expected behavior**
That models with types `List` and `Temporaral.DateTime` decode properly into the models

**Screenshots**
![Screenshot 2020-06-25 at 09 01 37](https://user-images.githubusercontent.com/24856374/85671757-7caf5f00-b6c2-11ea-8619-10553a03e8a8.png)
![Screenshot 2020-06-25 at 09 02 07](https://user-images.githubusercontent.com/24856374/85671821-8df86b80-b6c2-11ea-831a-bfffe95a75b9.png)

**Environment(please complete the following information):**
 - Amplify Framework Version: 1.0.1
 - Dependency Manager: Cocoapods
 - Swift Version : [e.g. 5.0]

**Device Information (please complete the following information):**
 - Device: Simulator

**Additional Context**
After further inspection I noticed the related user is also erroring with a more obscure reason:

![Screenshot 2020-06-25 at 09 05 41](https://user-images.githubusercontent.com/24856374/85672314-0e1ed100-b6c3-11ea-96a9-28fff2684b97.png)

**Comment by palpatim** [comment]
@imikemiller Sorry about the problems you're having with DataStore. We released [Amplify 1.0.2](https://github.com/aws-amplify/amplify-ios/releases/tag/v1.0.2) yesterday, which includes some DataStore bugfixes that might be relevant to your situation. Can you upgrade your installed Amplify pods (ensure your CocoaPods file doesn't pin to a specific Amplify version, then do a `pod update`); upgrade the Amplify CLI (`npm install -g @aws-amplify/cli`), and re-generate your model files with `amplify codegen models`. This will cause your model files to be regenerated with some reworked internals that support associated, non-model types better (which it looks like you might have in the `CheckinTag` type).

I have double-checked that Temporal and List are both supported as expected, even in 1.0.1, so I'm unclear what could be causing your issue, but let's start by getting the Amplify installation updated. 

If the problem persists, could you post your redacted schema as above, but include your definition for `CheckinTag`?

**Comment by palpatim** [comment]
Closing this due to inactivity. If the problem persists, please feel free to reopen with the additional information requested above.
